programs/escrow/src/lib.rs

```diff
@@ -51,33 +74,26 @@ pub mod escrow {
     /// - transfer funds back to giver
     /// - close the vault_token_account and escrow_account
     pub fn cancel(ctx: Context<Cancel>) -> ProgramResult {
-        let Cancel { giver, giver_token_account, vault_token_account, vault_authority, token_program, escrow_account, .. } = ctx.accounts;
```

Comment:
I just changed this - so that there wouldn't be a mutable borrow for the sake of the example code below.